src/main/java/uk/gov/hmcts/reform/et/syaapi/service/pdf/ET3FormService.java

```diff
@@ -0,0 +1,196 @@
+import uk.gov.hmcts.reform.et.syaapi.service.CaseDocumentService;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+import java.util.ArrayList;
+
+import static uk.gov.hmcts.ecm.common.constants.PdfMapperConstants.PDF_TYPE_ET3;
+import static uk.gov.hmcts.ecm.common.service.pdf.et3.ET3FormConstants.ET3_FORM_CLIENT_TYPE_RESPONDENT;
+import static uk.gov.hmcts.ecm.common.service.pdf.et3.ET3FormConstants.SUBMIT_ET3_CITIZEN;
+import static uk.gov.hmcts.reform.et.syaapi.constants.DocumentCategoryConstants.ET3_PDF_DOC_CATEGORY;
+import static uk.gov.hmcts.reform.et.syaapi.constants.EtSyaConstants.ENGLISH_LANGUAGE;
+import static uk.gov.hmcts.reform.et.syaapi.constants.EtSyaConstants.ET3;
+import static uk.gov.hmcts.reform.et.syaapi.constants.EtSyaConstants.PDF_FILE_TIKA_CONTENT_TYPE;
+import static uk.gov.hmcts.reform.et.syaapi.constants.EtSyaConstants.WELSH_LANGUAGE;
+import static uk.gov.hmcts.reform.et.syaapi.constants.ManageCaseRoleConstants.ET3_RESPONSE_LANGUAGE_PREFERENCE_WELSH;
+import static uk.gov.hmcts.reform.et.syaapi.constants.ResponseConstants.ET3_FORM_DOCUMENT_TYPE;
+import static uk.gov.hmcts.reform.et.syaapi.constants.ResponseConstants.UNABLE_TO_UPLOAD_DOCUMENT;
+import static uk.gov.hmcts.reform.et.syaapi.service.pdf.PdfUploadService.createPdfDocumentDescriptionFromCaseData;
+import static uk.gov.hmcts.reform.et.syaapi.service.pdf.PdfUploadService.sanitizePartyName;
+
+
+@Slf4j
+@Service
+@RequiredArgsConstructor()
+public class ET3FormService {
+    private final PdfService pdfService;
+    private final CaseDocumentService caseDocumentService;
+    private final IdamClient idamClient;
+
+    @Value("${pdf.et3English}")
+    public String et3EnglishPdfTemplateSource;
+    @Value("${pdf.et3Welsh}")
+    public String et3WelshPdfTemplateSource;
+
+    private static final String STRING_UNDERSCORE = "_";
+
+    public static String createET3PdfDocumentNameFromCaseData(CaseData caseData,
+                                                               String documentLanguage,
+                                                               UserInfo userInfo,
+                                                               RespondentSumTypeItem selectedRespondent) {
+        String respondentName = getRespondentNameBySelectedRespondent(selectedRespondent, userInfo);
+        String caseNumber = StringUtils.EMPTY;
+        if (ObjectUtils.isNotEmpty(caseData) && StringUtils.isNotBlank(caseData.getEthosCaseReference())) {
+            caseNumber = caseData.getEthosCaseReference().replace("/", StringUtils.EMPTY);
+        }
+        return ET3 + STRING_UNDERSCORE
+            + sanitizePartyName(respondentName)
+            + STRING_UNDERSCORE
+            + sanitizePartyName(caseNumber)
+            + STRING_UNDERSCORE
+            + (ENGLISH_LANGUAGE.equals(documentLanguage) ? StringUtils.EMPTY : STRING_UNDERSCORE)
+            + documentLanguage
+            + ".pdf";
```

Comment:
Not sure if we should be having underscores in the file name, spaces and dashes are preferred to format the document name

Reply:
Just to have file names without spaces. Generally it is accepted as a good practice to not to have spaces in file names. (https://superuser.com/questions/29111/what-technical-reasons-exist-for-not-using-space-characters-in-file-names) Normally also _ (underscores) are not advised to use because those file names may cause some conflicts in the command prompts. If there is a reason to use space, I can revert it back to space.

Reply:
Updated as 
return ET3 + StringUtils.SPACE + STRING_DASH + StringUtils.SPACE
            + sanitizePartyName(respondentName)
            + (ENGLISH_LANGUAGE.equals(documentLanguage)
            ? StringUtils.EMPTY
            : StringUtils.SPACE + STRING_DASH + StringUtils.SPACE + documentLanguage)
            + ".pdf";